Makefile

```diff
@@ -11,52 +12,59 @@ favicon:
 website:
 	LONG_CACHE=false npm run build
 
-# `website` is needed for the server deploys.
-deploy: website deploy-s0 deploy-s1 deploy-s2 deploy-gh-pages deploy-gh-pages-clean
+deploy: deploy-s0 deploy-s1 deploy-s2 clean-server-deploy deploy-gh-pages deploy-gh-pages-clean
 
-deploy-s0:
+deploy-s0: prepare-server-deploy push-s0
+deploy-s1: prepare-server-deploy push-s1
+deploy-s2: prepare-server-deploy push-s2
+
+prepare-server-deploy: website
 	# Ship a copy of the front end to each server for debugging.
 	# https://github.com/badges/shields/issues/1220
-	git add -f Verdana.ttf private/secret.json build/
-	git commit -m'MUST NOT BE ON GITHUB'
-	git push -f s0 HEAD:master
-	git reset HEAD~1
-	git checkout master
+	rm -rf ${SERVER_TMP}
+	git worktree prune
+	git worktree add -B production ${SERVER_TMP}
+	cp -r build ${SERVER_TMP}
+	git -C ${SERVER_TMP} add build/
+	git -C ${SERVER_TMP} commit -m '[DEPLOY] Add frontend for debugging'
+	cp private/secret-production.json ${SERVER_TMP}/private/secret.json
+	cp Verdana.ttf ${SERVER_TMP}
+	git -C ${SERVER_TMP} add private/secret.json Verdana.ttf
+	git -C ${SERVER_TMP} commit -m '[DEPLOY] MUST NOT BE ON GITHUB'
```

Comment:
That is a significant change in behavior. It feels like it does more than just use secret-production.json instead of secret.json. Could you explain this change further please?

Reply:
Without a working tree, the solution as far as I can see is to:

1. Back up the existing `private/secret.json`, if it exists
2. Copy `private/secret-production.json` to `private/secret.json`
3. Check it in
4. Restore the original `private/secret.json` afterward

That has more moving parts than I'd like, seems brittle, and also seems like it could result in accidental data loss.

By moving the branch manipulation into a separate working tree, `private/secret.json` can be copied for deployment without needing to preserve the original.

The other change is how the branching is handled, which I'll respond to below.